### ISSUE TITLE
[Bug]: ReturnTypeData.innerTypeName doesn't work with a return type being a Map

### ISSUE THREAD

**Issue by kylelutze** [Foso/Ktorfit#399]
### Ktorfit version

1.6.0

### What happened and how can we reproduce this issue?

if the return object of an API is a Map, like `Map<String, List<SomeOjbect>>` the generated code becomes
```
    return ktorfitClient.suspendRequest<Map<String, List<SomeOjbect>>, String,
        List<SomeOjbect>>(_requestData)!!
```
This doesn't work since now there are 3 values instead of 2 as per the interface. This seems to be related to how ReturnTypeData.innerTypeName works.

### What did you expect to happen?

A return type of a Map generates valid code

**Comment by Foso** [comment]
Hi @kylelutze thank your for reporting this!

**Comment by DCNick3** [comment]
I have a similar issue with a type like `Result<List<AttendanceSession>, BaamError>`. Seems like the generics support in general is a bit iffy.

This can be worked around by using type aliases: you alias the generic type to a simple name and then use it in the API

**Comment by Foso** [comment]
This is fixed with Ktorfit 2.0.0
